spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -104,8 +114,12 @@ public void setHeaderMapper(HeaderMapper<Map<String, String>> headerMapper) {
 	protected void doStart() {
 		super.doStart();
 
+		addToHealthRegistry();
```

Comment:
This is something I might be missing -- since the adapter will just delegate to the `SubscriberTemplate`, which will use `DefaultSubscriberFactory`, why do Spring Integration and Spring Cloud Stream layers manage their own health registration by way of `PubSubInboundChannelAdapter`? Wouldn't just instrumenting `DefaultSubscriberFactory` be enough?

Reply:
`PubSubInboundChannelAdapter` if used will wrap the MessageReceiver which is submitted to `DefaultSubscriberFactory` (through the `PubSubSubscriberTemplate`). 

https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/bf87624881d5287dd4bbba115eaedf90721f52c9/spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/integration/inbound/PubSubInboundChannelAdapter.java#L142-L160

In case of error, it will catch the error beforehand and it won't occur inside the receiver wrapped by `DefaultSubscriberFactory`. This way `DefaultSubscriberFactory` will always mark the subscriber as alive.

https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/1e2afb7f613bd914bf31a3701d12bea32d0a3f8c/spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/health/HealthTrackerRegistry.java#L37-L44

Based on the current structure, if `PubSubInboundChannelAdapter` is used for a subscription it would be responsible for the health checks thus `DefaultSubscriberFactory` is avoided for health check.
If `DefaultSubscriberFactory` is used directly then it will be responsible for the health checks.

https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/1e2afb7f613bd914bf31a3701d12bea32d0a3f8c/spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java#L236-L239

https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/1e2afb7f613bd914bf31a3701d12bea32d0a3f8c/spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java#L548-L555